### PR TITLE
Add setup check for pending bigint conversion

### LINKED ISSUE #12763: Add setup check for missing bigint conversion
There should be a setup check for https://docs.nextcloud.com/server/15/admin_manual/configuration_database/bigint_identifiers.html

### PR DESCRIPTION
**Pull request by MorrisJobke**
Requires:

* [x] #12821 (first commit in here will be gone once this is merged)

This contains multiple changes, that are best reviewed on a per commit basis:

* it fixes the problem with SQLite where primary fields can't be bigint and thus a conversion does not work where the warning would be shown all the time - see https://stackoverflow.com/a/18835967/520507
* it lists now the columns that need to be updated on the CLI before one needs to accept to run the command
* it shows the pending columns also in the settings check with a link to the documentation page: https://docs.nextcloud.com/server/15/admin_manual/configuration_database/bigint_identifiers.html
* fixes #12763 

This is how it looks like:

bildschirmfoto 2018-12-04 um 18 46 27

bildschirmfoto 2018-12-04 um 18 47 10

### CODE REVIEW

**Review by skjnldsv — APPROVED** [review]
Code make sense!
I did not even know we had `db:convert-filecache-bigint` :)

File: core/js/setupchecks.js [hunk 1]
```diff
@@ -324,6 +324,35 @@
 							type: OC.SetupChecks.MESSAGE_TYPE_INFO
 						})
 					}
+					if (data.recommendedPHPModules.length > 0) {
```
Comment (danielkesselberg) [inline core/js/setupchecks.js]: This belongs to https://github.com/nextcloud/server/pull/12821

**Comment by MorrisJobke** [comment]
@danielkesselberg I rebased this branch now and it is ready for review :)

**Comment by MorrisJobke** [comment]
> Code make sense!
> I did not even know we had `db:convert-filecache-bigint` :)

And this is the reason I added this 😆

**Review by ChristophWurst — APPROVED** [review]
Nice! Code looks good, thanks for the tip of reviewing commit per commit :nerd_face:

**Comment by danielkesselberg** [comment]
> TypeError: undefined is not an object (evaluating 'data.pendingBigIntConversionColumns.length') in core/js/setupchecks.js (line 9)

CI not happy 💥

**Comment by MorrisJobke** [comment]
> CI not happy 💥

Fixed 🙈

**Comment by ChristophWurst** [comment]
This one looks suspicious https://drone.nextcloud.com/nextcloud/server/13349/267 (though feature is unrelated)

The other failing acceptance tests seem to have timed out.

**Comment by MorrisJobke** [comment]
> This one looks suspicious https://drone.nextcloud.com/nextcloud/server/13349/267 (though feature is unrelated)

Looks fine and happened on other PRs as well.

**Pull request merged**